### PR TITLE
Update release-process.md

### PR DESCRIPTION
**Pull request by jasonblais**
Updates to release process:
 1. T-15: Last day to submit major features (to be enforced with Mattermost v3.5)
 2. T-14: Day when leads and PM area owners meet to discuss which major features will be cut from the release, if any.
 3. T-12: Last day to merge major features
 4. T-8/T-7: Largely unchanged
 5. T-5: Release testing finished by standup, worst bugs discussed with the team
 6. T-2/T-0 largely unchanged

Other updates:
 7. API changes added to changelog
 8. Marketing process started earlier, so a draft is ready for marketing lead by T-5
 9. Docs process started earlier to allow a couple extra days for delas
 10. Removed redundant steps for GitLab MR

### CODE REVIEW

**Comment by jasonblais** [comment]
Major updates to be reviewed in platform meeting, adding a WIP label until after the meeting is over

**Comment by it33** [comment]
Not urgent, propose additional tweaks: https://github.com/mattermost/docs/pull/389

File: source/process/release-process.md [hunk 1]
```diff
@@ -135,41 +170,43 @@ The final release is cut. If an urgent and important issue needs to be addressed
...
     - Update [MVP page](https://www.mattermost.org/mvp/) with the most valuable contributor of the release.
-    - Submit [GitLab MR](https://gitlab.com/gitlab-org/omnibus-gitlab/merge_requests/871/diffs) to take next Mattermost version in the Omnibus.
+    - Submit GitLab MR for `config/software/mattermost.rb` to take next Mattermost version in the Omnibus. The MR should include changes to the `default_version` and `md5` sum (see [example](https://gitlab.com/gitlab-org/omnibus-gitlab/merge_requests/933/diffs)
     - Update [Docker preview image to latest version](https://github.com/mattermost/mattermost-docker-preview/commit/1c34195d70b26bd4c82b5ef4fa0ebaf421096881).
 4. Marketing:
     - Finalize Mailchimp email blast
     - Finalize blog post for mattermost.org and set timer for 06:00 PDT on the day of release.
     - Queue Tweet announcement
-    - Find [GitLab release announcement merge request](https://gitlab.com/gitlab-com/www-gitlab-com/merge_requests) and post proposal for [GitLab Mattermost update text](https://gitlab.com/gitlab-com/www-gitlab-com/merge_requests/1633/diffs#efdba966a222d7071509e0dd4f39de5c9d7c1200_0_185)
```
Comment (it33) [inline source/process/release-process.md]: This isn't a redundant step, 
1. There's an MR to add the latest version of Mattermost to the latest GitLab release (above)
2. There needs to be a COMMENT by Mattermost on the MR for the GitLab to include a mention of Mattermost in the GitLab BLOG POST for their next release.

**Comment by it33** [comment]
Also, propose adding a step for PM to update direction (https://about.mattermost.com/direction/). Not sure on when, propose PM team decides.

**Comment by jasonblais** [comment]
Thanks @it33 for the all updates above, make sense. 

Roadmap updates are currently included in T+5 checklist.

@lfbrock Can you help update the PR based on
1) your discussion in leads on Friday regarding a) feature flags, b) when features should be merged and c) when decisions on what goes in/out should be made (I don't think I'm fully up to speed what was discussed there)
2) review feedback from @it33 in https://github.com/mattermost/docs/pull/389/files
3) help quarterback [GitLab release announcement merge request](https://github.com/mattermost/docs/pull/387#discussion_r75619012) for v3.3 and add it back to release process

File: source/process/release-process.md [hunk 2]
```diff
@@ -14,26 +14,47 @@ Notes:
...
     - Confirm changes to config.json in compatibility section of Changelog are written back to [settings documentation](http://docs.mattermost.com/administration/config-settings.html#configuration-settings)
     - Update [Upgrade Guide](http://docs.mattermost.com/administration/upgrade.html#upgrade-guide) for any steps needed to upgrade to new version
-    - Submit NOTICE.txt PR for any new libraries added from dev, if not added already
+    - Submit NOTICE.txt PR for any new libraries added from dev, if not added already. The following two files contain a list of dependancies:
+        - https://github.com/mattermost/platform/blob/master/webapp/package.json
+        - https://github.com/mattermost/platform/blob/master/glide.yaml
     - Create meta issue for release in GitHub (see [example](https://github.com/mattermost/platform/issues/3702)).
```
Comment (lfbrock) [inline source/process/release-process.md]: I think this should be under Code Complete after RC is cut?

**Comment by esethna** [comment]
Looks good, thanks LB for review and tweaks. +1

**Pull request merged**